lib/token-util.js

```diff
@@ -0,0 +1,46 @@
+var assign = require('lodash.assign')
+
+var Token // Token constructor from markdown-it
+
+var tokenUtil = module.exports = {}
+
+tokenUtil.set = function (TokenConstructor) {
+  Token = TokenConstructor
+}
+
+// set/replace or append to an attribute on a markdown-it Token
+tokenUtil.attr = function (token, attr, value, operation) {
+  var idx = token.attrIndex(attr)
+  if (idx < 0) {
+    token.attrPush([attr, value])
+  } else {
+    if (operation === 'append') {
+      token.attrs[idx][1] += ' ' + value
+    } else {
+      token.attrs[idx][1] = value
+    }
+  }
+}
```

Comment:
rebase onto the other, we're gonna land em together i think (unless you or anyone else has objections)